### ISSUE TITLE
Rewrite search system

### ISSUE THREAD

**Issue by ckniffen** [ripple/explorer#916]
## Goals

- Parallelize the search requests
- Show more detailed messages when search fails

## Current Flow

- Narrow down paths by regex
  - One possibility: redirect to page
  - More than one possibility: Make requests in order.  Ex. For a 64 character hex string lookup transaction and if that fails nft
    - On fail redirect to last thing attempted.

## New Flow

- Narrow down paths by regex
  - One possibility: redirect to page
  - More than one possibility: Make requests in parallel.
    - On failure of all display a message saying what types of items were searched for.
  - Upon a match redirect to matching page

## Details

Right now search largely narrows down the query by regex.  That was sufficient until NFT searching was added which matches the same criteria as transactions.  Amendments hashes match will also match as will most other searches involving hashes as they are often 64 characters.

When no results found show a message detailing all the things attempted and remain on current page.  This is a user experience change as it currently takes the user to a 404 page for the last type. This will address #544 where if there is not a transaction or an NFT it directs to the NFT page which is confusing.

## Future Work

- Add amendment search by both hash and name #915
- Add ledger search by hash #831
